automatic/kubernetes-cli/update.ps1

```diff
@@ -25,25 +25,46 @@ function global:au_SearchReplace {
 }
 
 function global:au_GetLatest {
+    # Only report the supported Kubernetes streams.
+    # NB Upstream Kubernetes only supports the last three minor versions.
+    #    However, they can also do pre-releases before a final version is
+    #    released, so we have to check for an extra changelogs/minor
+    #    version to be able to find all the supported streams.
+    # See https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md
+    $minor_versions_to_keep = 3
+
     $changelog_page = Invoke-WebRequest -Uri $changelogs -UseBasicParsing
-    #Only the latest three minor versions are supported: https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md
-    #However, there are prereleases before the version is released, so the latest four changelogs/minor versions need to be checked.
-    $minor_version_changelogs = $changelog_page.links | ? href -match "CHANGELOG-[\d\.]+\.md" | Select-Object -Expand href -First 4
-    
+    $minor_version_changelogs = $changelog_page.links `
+      | Where-Object href -match "CHANGELOG-(?<version>\d+\.\d+)\.md`$" `
+      | Select-Object -Expand href -First ($minor_versions_to_keep+1)
+
     $streams = @{}
-    
+
     $minor_version_changelogs | ForEach-Object {
-        $minor_version = ($_ -split "CHANGELOG-" | Select-Object -Last 1).trim(".md")
-        $minor_changelog_page = Invoke-WebRequest -UseBasicParsing -Uri ("https://github.com" + $_)
-        $url64 = $minor_changelog_page.links | ? href -match "/v(?<version>[\d\.]+)/kubernetes-client-windows-amd64.tar.gz" | Select-Object -First 1 -Expand href
+        if ($streams.Count -ge $minor_versions_to_keep) {
+          return
+        }
```

Comment:
We shouldn't restrict the amount of streams being added.

Please remove this if statement.

Reply:
Although we can leave the 3 limit restriction for now, this if statement should still be removed.

Reply:
I've removed it in the last push. The output still looks like:

```json
{
  "Streams": {
    "1.22": {
      "Version": "1.22.12",
      "ReleaseNotes": "https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md",
      "URL32": "https://dl.k8s.io/v1.22.12/kubernetes-client-windows-386.tar.gz",
      "URL64": "https://dl.k8s.io/v1.22.12/kubernetes-client-windows-amd64.tar.gz"
    },
    "1.23": {
      "Version": "1.23.9",
      "ReleaseNotes": "https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md",
      "URL32": "https://dl.k8s.io/v1.23.9/kubernetes-client-windows-386.tar.gz",
      "URL64": "https://dl.k8s.io/v1.23.9/kubernetes-client-windows-amd64.tar.gz"
    },
    "1.24": {
      "Version": "1.24.3",
      "ReleaseNotes": "https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md",
      "URL32": "https://dl.k8s.io/v1.24.3/kubernetes-client-windows-386.tar.gz",
      "URL64": "https://dl.k8s.io/v1.24.3/kubernetes-client-windows-amd64.tar.gz"
    }
  }
}
```

Should it contain all the all versions inside a given stream (although the function name is `au_GetLatest`)? Or should it only contain the latest (as its now)?

Reply:
Could be that something unrelated prevents additional streams from being added, I'll ignore that for now and come back to it later to check what is causing the issue.